apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/Version2CustomTypeAdapterToAdapter.kt

```diff
@@ -0,0 +1,73 @@
+package com.apollographql.apollo3.api
+
+import com.apollographql.apollo3.api.json.JsonReader
+import com.apollographql.apollo3.api.json.JsonWriter
+import kotlin.jvm.JvmStatic
+
+/**
+ * An [Adapter] that wraps an Apollo Android v2 style [CustomTypeAdapter], to ease migration from v2 to v3.
+ */
+@Deprecated("Used for backward compatibility with 2.x, use Adapter instead")
+class Version2CustomTypeAdapterToAdapter<T>(
```

Comment:
Could we make that internal? And only use it from deprecated functions?

Reply:
Definitely, I was not sure if it would make sense to keep it public, but I guess that would not be very useful indeed.

Reply:
Oops, actually `internal` doesn't work because `ApolloClient` is in `apollo-runtime` whereas this is defined in `apollo-api`. But maybe it would make more sense to move it to `apollo-runtime` actually? I thought to put it next to `Adapter` but it would probably be needed only when using the client, so... WDYT?

Reply:
A third option would be to use an `Optin` `ApolloInternal` annotation to make sure anyone using this knows it's not intended for public use?

Reply:
Ah that's a good idea! And could be useful in the future for similar situations 👍.  Let me have a 👀

Reply:
Added the annotation (also moved `Version2CustomTypeAdapterToAdapter` to the `internal` package)